design/UsageValidationRules.md

```diff
@@ -0,0 +1,69 @@
+## Definition of "resources being used"
+
+Resources in WebGPU can be used in different ways that are declared at resource creation time and interact with various validation rules.
+
+Buffers have the following usages (and commands inducing that usage):
+ - `VERTEX` for the `buffers` in calls to `WebGPUCommandEncoder.setVertexBuffers`
+ - `INDEX` for `buffer` in calls to `WebGPUCommandEncoder.setIndexBuffer`
+ - `INDIRECT` for `indirectBuffer` in calls to `WebGPUCommandEncode.{draw|drawIndexed|dispatch}Indirect`
+ - `UBO`, `RO_STORAGE` and `STORAGE` for buffers referenced by bindgroups passed to `setBindGroup`, with the usage corresponding to the binding's type.
+ - `TRANSFER_SRC` for buffers used as the copy source of various commands.
+ - `TRANSFER_DST` for buffers used as the copy destination of various commands.
+ - (Maybe [RO_]STORAGE_TEXEL and UNIFORM_TEXEL?)
+
+Textures have the following usages:
```

Comment:
Missing `TRANSFER_SRC` and `TRANSFER_DST`

Reply:
Done